lib/Serialization/DeserializeSIL.cpp

```diff
@@ -3108,8 +3118,12 @@ SILWitnessTable *SILDeserializer::readWitnessTable(DeclID WId,
   }
 
   // Deserialize Conformance.
+  auto maybeConformance = MF->readConformanceChecked(SILCursor);
+  if (!maybeConformance)
```

Comment:
could you add some comments why this error may happen? Otherwise LGTM.

Reply:
I've added documentation on the prototype for readWitnessTableChecked and on the only client readWitnessTables.